### PR TITLE
[UI/UX:Autograding] Distinguish Insertion/Deletion results

### PR DESCRIPTION
**Pull request by GO-viper7**
### Please check if the PR fulfills these requirements:

* [x] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
Students cannot specifically identify what's wrong with their autograding results as there are red marking on both expected and actual data.

### What is the new behavior?
Actual result gets red markings only when there's a need of deletion to get matched with expected result.
Expected result gets green markings only when there's a need of insertion in actual result to get matched with expected result.

**Light Theme :** 
image

**Dark Theme :**
image

**Black Theme :** 
image

### CODE REVIEW

**Review by ZheyuDeng — APPROVED** [review]
Colors look good to me. I put how the color looks in dark mode below so it's easier for other people to review.
image

**Review by williamjallen — COMMENTED** [review]
I'm going to move this back to "Work in Progress" since the tests still need to be fixed.  I think the changes look reasonable overall.  The failing colors.css CI check can be ignored, since the purpose of this PR was to add new colors for the diff viewer.

**Review by williamjallen — APPROVED** [review]
The code and basic functionality looks reasonable to me.  I did not test this extensively though, so it would be good to have others look at this as well.

File: site/public/css/diff-viewer.css [hunk 1]
```diff
@@ -8,24 +8,37 @@
...
 
 .highlight-hover {
     background-color: var(--standard-light-blue-gray) !important;
 }
 
-.highlight-char {
-    background-color: var(--focus-danger-red);
+.highlight-char-actual {
+    background-color: var(--diff-char-background-actual);
+    font-weight: 900;
```
Comment (ziesski) [inline site/public/css/diff-viewer.css:36]: I don't think we agreed on adding the boldness for highlighted char.

File: site/public/css/diff-viewer.css [hunk 2]
```diff
@@ -8,24 +8,37 @@
...
+}
+
+.highlight-char-expected {
+    background-color: var(--diff-char-background-expected);
+    font-weight: 900;
```
Comment (ziesski) [inline site/public/css/diff-viewer.css:41]: and this too.

**Review by ziesski — CHANGES_REQUESTED** [review]

File: site/public/css/diff-viewer.css (hunk 1 above)
Comment (GO-viper7) [reply site/public/css/diff-viewer.css:36]: The demo I gave the other day included making highlighted characters bold as well. 
May be its my screenshot fault that you missed it. 
We can discuss in today's meet :)

**Review by ziesski — APPROVED** [review]
After updated images, I think it is in good shape, I like it better without the bold font.

**Comment by ziesski** [comment]
#10762 was meant to fix some of this problem, line hight not matching the empty rows. It looks like your latest commit is reverting the change, could you explain why we need to revert the changes? @GO-viper7

**Comment by GO-viper7** [comment]
> #10762 was meant to fix some of this problem, line hight not matching the empty rows. It looks like your latest commit is reverting the change, could you explain why we need to revert the changes? @GO-viper7

I wanted to discuss this in today's meet
So, yes it indeed solves the issue which Zheyu mentioned but makes UI for the tables a bit weird (as u can see  `|` in tables need to be as close as possible)
You can see the example right here, compare it with the pics above and please comment your views.
image

**Comment by bmcutler** [comment]
I tweaked the dark green color for light mode.
New screenshot, in Mac Chrome:
light_chrome

in Mac Safari:
light_safari

I note that the vertical spacing looks different -- there is more vertical space in Safari and it seems unnecessary.  That can/should be addressed in a future PR.

**Review by bmcutler — APPROVED** [review]
This PR is expected to update colors.css and add colors, so I am merging it with the test failure on the colors.css check.

**Pull request merged**